### PR TITLE
Use natural numbers for test, experiment and trial IDs

### PR DESCRIPTION
**Pull request by jeppe-style**
Fixes https://github.com/benchflow/benchflow/issues/290

### CODE REVIEW

File: benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java [hunk 1]
```diff
@@ -185,34 +188,39 @@ private void handleReExecuteTrial(String experimentID) {
 
     logger.info("handleReExecuteTrial: " + experimentID);
 
-    ReExecuteTrialTask reExecuteTrialTask = new ReExecuteTrialTask(experimentID);
+    try {
 
-    Future<RunStatus> future = experimentTaskExecutorService.submit(reExecuteTrialTask);
+      String trialID = experimentModelDAO.getLastExecutedTrialID(experimentID);
```
Comment (VincenzoFerme) [inline benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java:193]: You should know the trial when you access this method, without the need of querying the model.

File: benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/BenchFlowTestNumber.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 
   @Id private String testIdentifier;
 
-  private Long counter = 0L;
+  private Long counter = 1L;
```
Comment (VincenzoFerme) [inline benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/BenchFlowTestNumber.java:17]: How do you use this number to increment the test number if the user has already executed a test with the same name?

**Review by VincenzoFerme — CHANGES_REQUESTED** [review]

File: benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/BenchFlowTestNumber.java (hunk 2 above)
Comment (jeppe-style) [reply benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/models/BenchFlowTestNumber.java:17]: In the constructor you can see that it takes a a `userName` and `testName` as input meaning that there will be one `BenchFlowTestNumber` object stored in the DB per combination. In the [TestModelDAO](https://github.com/benchflow/benchflow/blob/408297ca24ea6f12d257f9c4ca33b55ea3e236ef/benchflow-test-manager/application/src/main/java/cloud/benchflow/testmanager/services/internal/dao/BenchFlowTestModelDAO.java#L43) class you can see how the test number is generated.

File: benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java (hunk 1 above)
Comment (jeppe-style) [reply benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java:193]: yes we need to rework this according to the updates we made to the lifecycle.

Comment (VincenzoFerme) [reply benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java:193]: Can you link the issue?

Comment (jeppe-style) [reply benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/ExperimentTaskController.java:193]: here is the issue https://github.com/benchflow/benchflow/issues/347

**Review by VincenzoFerme — APPROVED** [review]

**Pull request merged**